File: YeChanSong_코딩테스트_연습/DFS_BFS/타겟넘버.py
import sys

def dfs(idx,acc,numbers,target,sign):
    
    if idx == len(numbers)-1:
        if sign == '+':
            if acc + numbers[idx] == target:
                return 1
            else:
                return 0
        else:
            if acc - numbers[idx] == target:
                return 1
            else:
                return 0
    
    tmp = acc
    if sign == '-':
        tmp -= numbers[idx]
    else:
        tmp += numbers[idx]
    
    cnt = 0
    cnt += dfs(idx+1,tmp,numbers,target,'-')
    cnt += dfs(idx+1,tmp,numbers,target,'+')
    
    return cnt



def solution(numbers, target):
    answer = 0
    answer += dfs(0,0,numbers,target,'-')
    answer += dfs(0,0,numbers,target,'+')
    return answer

File: YeChanSong_코딩테스트_연습/DFS_BFS/타겟넘버.java
import java.util.*;

class Solution {
    static int answer = 0, lim;
    public int solution(int[] numbers, int target) {
        lim = numbers.length;
        dfs(0,0, target, numbers);
        return answer;
    }
    public void dfs(int idx, int sum, int target,int[] numbers){
        if(idx == lim){
            if(sum == target)
                answer++;
            return;
        }
        dfs(idx+1,sum+numbers[idx],target,numbers);
        dfs(idx+1,sum-numbers[idx],target,numbers);
        
    }
    
}